README.md
# GeoActivity - customizable tools/armor,ore quadrupling and more

![Image](http://i.imgur.com/0KtR3do.png)

## Downloads for the latest versions can be found on [CurseForge](http://minecraft.curseforge.com/projects/geoactivity).

## Features
- 3 new coal types
- new armor/tools sets that can be customized
- new blocks
- new machines
- Chemistry
- RF powered tools
 
## General Information
This mod adds 3 new different coal ores to the game:


Anthracite Ore - spawns on layers 18-38 in veins of maximum 3 blocks, more difficult to find

Bituminous Ore - spawns on layers 34-48 in veins of maximum 3 blocks, not very hard to find

Lignite Ore - spawns on layer 46-64 in veins of maximum 3 blocks, somewhat common


The mod adds a variety of modular tools that can be upgrade with Perks.There are also tools that act like a pickaxe and shovel in one and a tools which is a pick,shovel,sword,axe and hoe in one.Because they are modular, that means that you can put any Perk you want on it.

Perks range from Wider Radius, Fortune to No drop and Auto-Smelt.There are currently over 10 Perks available.
 
We also have implement a chemistry system, with atoms and molecules.Currently there are 19 elements that this mod adds.These elements are used in a Chemistry Machine to craft various items.Elements are extracted from items using the Coal Refiner.
 
There is also a multiblock furnace that can quadruple your resources if used correctly.Be sure to check it out!
 
The mod is compatible with NEI, so be sure to have that installed if you want to browse the recipes available for each machine.
To install, just place the .zip file in the "mods" folder in your .minecraft folder.

## Mod Showcase (v1.4)
https://www.youtube.com/watch?v=hbpYyVw6cqk



##### This mod is licensed under the [Creative Commons Attribution-NonCommercial-ShareAlike 4.0 International license](http://creativecommons.org/licenses/by-nc-sa/4.0/).


Changelog.txt
Version 2.3.1:
Changes:
- update to MC 1.9.4 and Forge 1.9.4-12.17.0.1976

Version 2.3:
Changes:
- ALPHA VERSION, expect bugs
- updated to MC 1.9 and Forge 1.9-12.16.0.1865

Version 2.2.3:
Fixes:
- fixed crash on client side
- various other fixes
- updated to official RF API for MC 1.8

Version 2.2.2:
Changes:
- added JEI support for Crafting Machine, Atom Extractor, Coal Refiner, Advanced Coal Refiner and Chemistry Machine
- added JEI descriptions for multiblocks such as Chemistry Machine and Advanced Ore Refiner
- nerfed experience output of Coal Refiner recipes
Fixes:
- fixed server side crash
- fixed Advanced Armor not absorbing damage properly
- fixed Advanced Ore Refiner not working with certain Ores

Version 2.2.1:
Changes:
- HUGE refactoring of package names and GUI handler
- added JEI support for the Crafting Machine recipes, with more machines to come in the future

Version 2.2:
Changes:
- updated to MC 1.8.9 and Forge 11.15.1.1722
Fixes:
- small fixes for Redstone Tools not mining properly

Version 2.1.1:
Fixes:
- fixed coal ores not dropping the proper item
- other small bug fixes

Version 2.1:
Changes:
- updated to recommended Forge 11.14.3.1450
- added Geothermal Generator;needs to be placed in lava pools;generates 4 RF/t per lava block;works better at lower altitudes;scans for lava blocks recursively up to a distance of 25 meters from the generator;
outputs power to bottom/top only and searches for lava blocks starting from its sides only;has a 40% to turn lava into obsidian in a 3x3x3 cube centered on it, so it will eventually stop working;
- added Thermic Melter;transforms obsidian into lava in a 5x5x5 cube centered on it;uses 10 RF/t passively and 50k RF per obsidian block;
Fixes:
- implemented proper interfaces for machines that use RF
- fixed Advanced Redstone BattleMiner GUI and added detailed power information to the power bar

Version 2.0:
Changes:
- updated to MC 1.8 and Forge 11.14.1.1334
- increased Advanced Redstone BattleMiner energy capacity to 5 million RF and transfer rate to 20k RF/t
- Chemistry Machine DOES NOT give back coal ore the element containers
- ToolCharger can now charge RF tools with up to 20k RF/t and can hold 10 million RF
Fixes:
- LOTS of bug fixes
- performance improvements

Version 1.6:
Changes:
- updated to MC 1.7.10 and Forge 10.13.0.1199
- added an Advanced Redstone BattleMiner, an RF powered tool, more expensive than the Redstone BattleMiner, that can be upgraded with up to 12 perks;it also supports up to 4 Wider Mining Radius perks 
which can increase the radius to 3x3x1, 5x5x1, 7x7x2 and 9x9x3
- added a Thermal Hardened Brick that is used together with the Thermic Generator to produce RF;place the bricks adjacent to the generator and for each brick it creates 20RF/t;power can only
be extracted through the bottom 
- added a Tool Charger that can accept up to 10000 RF/t and can only charge RF powered Redstone Tools

Version 1.5:
Changes:
- updated to Forge 10.12.1.1060
- added more useful recipes to the Chemistry Machine
Fixes:
- added back compatibility with NEI

Version 1.4:
Changes:
- updated to Forge 10.12.0.1047
- Advanced Tools can now be enchanted;enchantments do not stack with perks of the same type or other perks(like fortune enchant with auto-smelt)
like they do in the Redstone Tools
- decreased the enchantability of Advanced Tools and nerfed fuel values since they are pretty OP when they are enchanted
- buffed fuel values in Redstone Tools
- added config options for Ore Generation
- added Drop Teleportation Perk that can be bound to an inventory and when placed in a tool, drops will get automatically inserted into that
inventory and will not be dropped on the ground(unless there is no space left in the inventory)
- added Bypass Armor Perk which makes 1/3 of the tool damage bypass armor
- added Advanced Armor which can be customized with perks and function similar to Advanced Tools
- added General Protection Perk, Blast Protection Perk, Fire Protection Perk, Magic Protection Perk, Projectile Protection Perk, No Fall Damage Perk, Respiration Perk for armor
Fixes:
- tweaked tools damages

Version 1.3:
Changes:
- updated to MC 1.7.2 and Forge 10.12.0.1034
- nerfed the Coal Refiner so that the Advanced Coal Refiner is useful
Fixes
- small bug fixes and improvements

Version 1.2
Changes:
- changed some recipes for better balance
- added block of coal as fuel source and changed fuel burn times in the Crafting Machine
Fixes:
- more tool bug fixes

Version 1.1:
Fixes:
- tools work properly now
- LOTS of bug fixes

Version 1.0:
Changes:
- removed Crafting Machine I and moved it's recipes to the vanilla workbench
- removed Magma Ore and nerfed the generation of the coal ores
- the new coal ores can no longer be used as fuel
- coal exchange recipes were removed from the Crafting Machine
- coal can be crafted with 4 graphite in a square
- diamonds have a new recipe
- Coal Refiner has some new recipes
- coal ores give double output if they are put in the Coal Refiner
- new Advanced Ore Refiner, a complicated multiblock structure that acts like a furnace, doubling your output with a chance of tripling it or tripling your output 
with a chance of quadrupling it if you use a certain special item;works on ores that drop themselves only
- new Auto Stone Builder that can auto smelts/2x2 craft(or both) one block
into a specific block and then place that block in the world(useful for
transforming cobblestone into stone/stone bricks directly;has a 64 blocks 
internal capacity;uses coal to recharge)
- new Redstone Miner(pick+shovel) and Redstone BattleAxe(sword+axe+hoe)
(both can be upgraded with up to 4 perks and have 4 slots for holding fuel)
- new Advanced Redstone BattleAxe that acts like a pick,shovel,axe,hoe and sword
in one tool(can be upgraded with up to 8 perks)
IMPORTANT:Do not apply perks for these tools in the Crafting Machine!They will not work and will waste your perk!

- Reinforced Tools can no longer mine obsidian

src/main/java/geoactivity/common/GAConfig.java
package geoactivity.common;

import java.io.File;

import net.minecraftforge.common.config.Configuration;
import net.minecraftforge.fml.common.event.FMLPreInitializationEvent;

public class GAConfig 
{
	public int ligniteBlocks;
	public int ligniteMinY;
	public int ligniteMaxY;
	public int ligniteSize;
	
	public int bituminousBlocks;
	public int bituminousMinY;
	public int bituminousMaxY;
	public int bituminousSize;
	
	public int anthraciteBlocks;
	public int anthraciteMinY;
	public int anthraciteMaxY;
	public int anthraciteSize;
	
	public GAConfig(FMLPreInitializationEvent event)
	{
		Configuration config = new Configuration(new File(event.getModConfigurationDirectory()+"/GeoActivity", "geoactivity.cfg"));
		config.load();
		
		oreConfig(config, "Ore Generation.");
		
		config.save();
	}
	
	private void oreConfig(Configuration config, String baseCategory)
	{
		//Lignite Ore
		ligniteBlocks = config.get(baseCategory + "Lignite Ore", "blocks/chunk", 14).getInt(14);
		ligniteMinY = config.get(baseCategory + "Lignite Ore", "minY", 46).getInt(46);
		ligniteMaxY = config.get(baseCategory + "Lignite Ore", "maxY", 64).getInt(64);
		ligniteSize = config.get(baseCategory + "Lignite Ore", "max vane size", 3).getInt(3);
		
		//Bituminous Ore
		bituminousBlocks = config.get(baseCategory + "Bituminous Ore", "blocks/chunk", 14).getInt(14);
		bituminousMinY = config.get(baseCategory + "Bituminous Ore", "minY", 34).getInt(34);
		bituminousMaxY = config.get(baseCategory + "Bituminous Ore", "maxY", 48).getInt(48);
		bituminousSize = config.get(baseCategory + "Bituminous Ore", "max vane size", 3).getInt(3);
		
		//Anthracite Ore
		anthraciteBlocks = config.get(baseCategory + "Anthracite Ore", "blocks/chunk", 14).getInt(14);
		anthraciteMinY = config.get(baseCategory + "Anthracite Ore", "minY", 18).getInt(18);
		anthraciteMaxY = config.get(baseCategory + "Anthracite Ore", "maxY", 38).getInt(38);
		anthraciteSize = config.get(baseCategory + "Anthracite Ore", "max vane size", 3).getInt(3);
	}
}
